lib/manifest-parser.ts

```diff
@@ -36,11 +36,37 @@ export function getDependencyNamesFrom(
   );
 }
 
-function devDependenciesFrom(manifest: PoetryManifestType) {
-  return Object.keys(manifest.tool.poetry['dev-dependencies'] || []);
+const getGroupDevDepNames = (
+  obj: GroupProperty,
+  arr: string[] = [],
+): string[] => {
+  for (const key in obj) {
+    if (!obj[key] || typeof obj[key] !== 'object') {
+      arr.push(key);
+      break;
+    }
+    getGroupDevDepNames(obj[key] as GroupProperty, arr);
+  }
+
+  return arr;
+};
```

Comment:
I was presuming having to recursively check very nested objects (I initially thought you could have `manifest.tool.poetry.group.some.nested.group.name` but seems you can only have `manifest.tool.poetry.group.name`.